omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,123 @@
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+		}
+		
+		return description;
+  }
+  
+  @Override
+  public DelegatingResourceDescription getCreatableProperties() {
+      DelegatingResourceDescription description = new DelegatingResourceDescription();
+      description.addProperty("identifierType");
+      description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+      return description;
+  }
+
+  @Override
+  public DelegatingResourceDescription getUpdatableProperties() {
+      DelegatingResourceDescription description = new DelegatingResourceDescription();
+      description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+      return description;
+  }
+
+  @Override
+  public AutoGenerationOption newDelegate() {
+    return new AutoGenerationOption();
+  }
+
+  @Override
+  public AutoGenerationOption save(AutoGenerationOption autoGenerationOption) {
+    return Context.getService(IdentifierSourceService.class).saveAutoGenerationOption(autoGenerationOption);
+  }
+
+  @Override
+  public AutoGenerationOption getByUniqueId(String uniqueId) {
+    return Context.getService(IdentifierSourceService.class).getAutoGenerationOption(Integer.parseInt(uniqueId));
+  }
+
+  @Override
+  public void purge(AutoGenerationOption delegate, RequestContext context) throws ResponseException {
+    Context.getService(IdentifierSourceService.class).purgeAutoGenerationOption(delegate);
+  }
+
+  @Override
+  public void delete(AutoGenerationOption delegate, String id, RequestContext context) throws ResponseException {
+    throw new ResourceDoesNotSupportOperationException("Deleting of AutoGenerationOption is not supported");
+  }
+
+  @Override
+  protected PageableResult doGetAll(RequestContext context) throws ResponseException {
+    IdentifierSourceService iss = Context.getService(IdentifierSourceService.class);
+    List<AutoGenerationOption> optionMap = new ArrayList<AutoGenerationOption>();
+    List<PatientIdentifierType> identifierTypes = new ArrayList<PatientIdentifierType>();
+    Map<PatientIdentifierType, List<IdentifierSource>> availableSources = iss.getIdentifierSourcesByType(false);
+
+    for (PatientIdentifierType pit : Context.getPatientService().getAllPatientIdentifierTypes()) {
+      List<AutoGenerationOption> options = iss.getAutoGenerationOptions(pit);
+
+              if (options != null && options.size() > 0) {
+              optionMap.addAll(options);
+              }
+
+              if (availableSources.get(pit) != null && availableSources.get(pit).size() > 0) {
+                  identifierTypes.add(pit);
+              }
+          }
+		return new AlreadyPaged<AutoGenerationOption>(optionMap, context);
```

Comment:
Didn't we agree to pass in a list instead of map in the above?

Reply:
I changed the `optionMap` from `map` to `list`. Is there a better way to create a list than using `new ArrayList`?

Reply:
Then your variable name is misleading.